packages/dev/core/src/Misc/uvprojector.ts

```diff
@@ -0,0 +1,136 @@
+                pt2 = new Vector3().fromArray(pt2, 0);
+                pt3 = new Vector3().fromArray(pt3, 0);
+                convertToArray = true;
+            }    
+            const normal = Vector3.Zero();
+            Vector3.CrossToRef(pt2.subtract(pt1), pt3.subtract(pt1), normal)
+            normal.normalize();
+
+            if(convertToArray){
+                return normal.asArray();
+            }else{
+                return normal;
+            }
+        }
+
+
+        function predicate(mesh: AbstractMesh): boolean{
+            if (mesh == reference){
+                return true;
+            }
+            return false;
+        }
+        
+        //use ray to project the uv data from the reference mesh to the target mesh
+        for(let i = 0; i < indices.length; i++){
+    
+            if(i % 3 == 0){
+                let rayDir = new Vector3(0, 1, 0);
+    
+                let faceDir = getFaceNormal(vec[indices[i]], vec[indices[i + 1]], vec[indices[i + 2]]);
+                faceDir.normalize();
+                faceDir = new Vector3(Math.abs(faceDir.x), Math.abs(faceDir.y), Math.abs(faceDir.z))
+                
+                if(faceDir.x > faceDir.y && faceDir.x > faceDir.z){
+                    rayDir = new Vector3(1, 0, 0);
+                }else if(faceDir.y > faceDir.x && faceDir.y > faceDir.z){
+                    rayDir = new Vector3(0, 1, 0);
+                }else if(faceDir.z > faceDir.x && faceDir.z > faceDir.y){
+                    rayDir = new Vector3(0, 0, 1);
+                }
+    
+                rayDir = rayDir.normalize();
+                const rayLength = rayDir.length() * Math.abs(_width * _height * _depth);
+    
+                const faceVerts = [vec[indices[i]], vec[indices[i + 1]], vec[indices[i + 2]]];
+                const trigInd = [indices[i], indices[i + 1], indices[i + 2]]
+                
+                for(let n = 0; n < faceVerts.length; n++){
+                    
+                    let id = trigInd[n] || 0;
+                    id *= 2;
+    
+                    const origin = new Vector3(faceVerts[n].x, faceVerts[n].y, faceVerts[n].z);
+                    let ray =  new Ray(origin, rayDir, rayLength);
+                    let hit = scene.pickWithRay(ray, predicate);
+                    
+                    if(!hit?.pickedMesh){
+                        ray = new Ray(origin, rayDir.scale(-1), rayLength);
+                        hit = scene.pickWithRay(ray, predicate);
+                    }  
+    
+                    if(!hit?.pickedMesh){ continue; }
+    
+                    const uvh = hit.getTextureCoordinates();
+    
+                    uvs[id] = uvh?.x;
+                    uvs[id + 1] = uvh?.y;
+                }
+            }
+        };
+        target.setVerticesData(VertexBuffer.UVKind, uvs);
+    }
+    /**
+     * Get class name
+     * @returns "UvProjector"
+     */
+     public getClassName(): string {

```

Comment:
I don't think we need this method as the function is a static function.

If we keep it, it should return `UVProject`.